### PR TITLE
BUG/PERF: sparse min/max don't densify

### PR DESCRIPTION
**Pull request by mzeitlin11**
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

This now follows  the structure of other `SparseArray` reductions like `sum` more closely, though there is some added complexity here to avoid taking the `min/max` of an empty array, which would raise when we want to return the corresponding missing value instead.

Benchmarks:
```
       before           after         ratio
     [5deec13d]       [f3eea313]
     <master>         <perf_sparse_min_max>
-      4.05±0.3ms       11.6±0.3μs     0.00  sparse.MinMax.time_min_max('max', 0.0)
-      4.49±0.4ms       11.6±0.2μs     0.00  sparse.MinMax.time_min_max('min', 0.0)
-      4.62±0.7ms       10.6±0.2μs     0.00  sparse.MinMax.time_min_max('max', nan)
-      5.37±0.7ms       10.3±0.3μs     0.00  sparse.MinMax.time_min_max('min', nan)
```

### CODE REVIEW

File: pandas/core/arrays/sparse/array.py [hunk 1]
```diff
@@ -1433,23 +1433,70 @@ def mean(self, axis=0, *args, **kwargs):
...
+        if len(valid_vals) > 0:
+            sp_min_max = getattr(valid_vals, kind)()
+
+            # If a non-null fill value is currently present, it might be the min/max
+            if has_nonnull_fill_vals:
+                func = max if kind == "max" else min
+                return func(sp_min_max, self.fill_value)
+            else:
+                return sp_min_max
         else:
```
Comment (jreback) [inline pandas/core/arrays/sparse/array.py:1495]: might has well just do elif else

**Review by jreback — CHANGES_REQUESTED** [review]
small comment, ping on green.

File: pandas/core/arrays/sparse/array.py (hunk 1 above)
Comment (mzeitlin11) [reply pandas/core/arrays/sparse/array.py:1520]: thanks, updated

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @mzeitlin11

**Pull request merged**